lib/github-pages-health-check/domain.rb

```diff
@@ -177,9 +176,17 @@ def pointed_to_github_pages_ip?
         a_record? && CURRENT_IP_ADDRESSES.include?(dns.first.address.to_s)
       end
 
-      # Is the domain's first response an A record to the new primary IPs?
-      def pointed_to_new_primary_ips?
-        a_record? && NEW_PRIMARY_IPS.include?(dns.first.address.to_s)
+      # Are any of the domain's A records pointing elsewhere?
+      def non_github_pages_ip_present?
+        return unless dns?
+        a_records = dns.select { |answer| answer.type == Dnsruby::Types::A }
+
+        a_records.all? do |answer|
+          return true unless github_pages_ip?(answer.address.to_s)
```

Comment:
To be explicit, you mean to return from the function here right? Perhaps we could use `all?`'s built-in flow instead of returning early from the function? What about simply:

`a_records.any? { |answer| !github_pages_ip?(answer.address.to_s) }`

> True if any A records are not GitHub Pages IPs, false otherwise